cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -454,6 +455,36 @@ where
         Ok(())
     }
 
+    fn generate_switch(&mut self, builder: &mut FunctionBuilder) -> Result<()> {
+        let _type = *self.u.choose(
+            &[
+                I8, I16, I32, I64, // TODO: I128
+            ][..],
+        )?;
+        let switch_var = self.get_variable_of_type(_type)?;
+        let switch_val = builder.use_var(switch_var);
+
+        let valid_blocks = self.generate_valid_jumptable_target_blocks();
+        let default_block = *self.u.choose(&valid_blocks[..])?;
+
+        // Build this into a HashMap since we cannot have duplicate entries.
+        let entry_count = self.param(&self.config.switch_target_blocks)?;
+        let mut entries = HashMap::with_capacity(entry_count);
+        for _ in 0..entry_count {
+            let index = self.u.arbitrary()?;
```

Comment:
`Switch` is designed to work for any kind of pattern. It should work just as well for switches like 1..5, 7..12 as for those like 1, 7, 33, 91. The former will use a branch and then two jump tables, while the later will only use branches. I presume the current code is biased towards the later option, right?

Reply:
Yeah, Its going to be very unlikely that the fuzzer will generate that. I'll add a bias for adding contiguous ranges.